extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -320,29 +384,35 @@ class CornerstoneViewportService implements IViewportService {
 
     this.viewportsDisplaySets.set(viewport.id, [displaySetInstanceUID]);
 
-    let initialImageIndexToUse = initialImageIndex;
+    let initialImageIndexToUse =
+      presentation?.initialImageIndex ?? initialImageIndex;
 
     if (!initialImageIndexToUse) {
       initialImageIndexToUse =
         this._getInitialImageIndexForStackViewport(viewportInfo, imageIds) || 0;
     }
 
-    const { voi, voiInverted } = displaySetOptions[0];
-    const properties = {};
-    if (voi && (voi.windowWidth || voi.windowCenter)) {
-      const { lower, upper } = csUtils.windowLevel.toLowHighRange(
-        voi.windowWidth,
-        voi.windowCenter
-      );
-      properties.voiRange = { lower, upper };
-    }
+    const properties = presentation?.properties || {};
+    if (!presentation?.properties) {
+      const { voi, voiInverted } = displaySetOptions[0];
+      if (voi && (voi.windowWidth || voi.windowCenter)) {
+        const { lower, upper } = csUtils.windowLevel.toLowHighRange(
+          voi.windowWidth,
+          voi.windowCenter
+        );
+        properties.voiRange = { lower, upper };
+      }
 
-    if (voiInverted !== undefined) {
-      properties.invert = voiInverted;
+      if (voiInverted !== undefined) {
+        properties.invert = voiInverted;
+      }
     }
 
     viewport.setStack(imageIds, initialImageIndexToUse).then(() => {
       viewport.setProperties(properties);
+      const { zoom, pan } = presentation || {};
+      if (zoom && !isNaN(zoom)) viewport.setZoom(zoom);
+      if (pan && !isNaN(pan[0]) && !isNaN(pan[1])) viewport.setPan(pan);
```

Comment:
we better refactor the `setPresentation` so that these are not repeated, you have the same logic in the `setPresentation` too